learn/get-started/work-with-data-using-queries-in-ballerina.md

```diff
@@ -121,23 +165,60 @@ public function findCountriesByHighestNoOfDeaths(table<CovidEntry> dataTable, in
     return countriesWithDeaths;
 }
 ```
-The `findCountriesByHighestNoOfDeaths` function uses query expression to find the top three countries with the highest COVID-19 deaths. The `order by` clause is used to sort the records in the table in `descending` order and the `limit` clause to limit the number of output records of the query to `n`. As the result, the query produces an array of tuples of type `[string, decimal]`. Each tuple contains the country name and the number of reported deaths. The produced array is in descending order by the number of deaths.
 
-Let's call the `findCountriesByHighestNoOfDeaths` function from within the `main` function to find the top three countries by the number of deaths.
+In this code,
+- You use another query to sort and retrieve a *limit*ed number of records from the table.
+- The `findCountriesByHighestNoOfDeaths` function uses a query expression to find the top three countries with the highest COVID-19 deaths. 
+- The `order by` clause is used to sort the records in the table in `descending` order and the `limit` clause to limit the number of output records of the query to `n`. 
+- The query produces an array of tuples of type `[string, decimal]` as the result. Each tuple contains the country name and the number of reported deaths. 
+- The produced array is in descending order by the number of deaths.
+
+### Update the `main` function for sorting
+
+To call the `findCountriesByHighestNoOfDeaths` function from within the `main` function to find the top three countries by the number of deaths, add the code below to the `main` function of the `main.bal` file.
+
+```ballerina
+    [string, decimal][] countriesWithDeaths = findCountriesByHighestNoOfDeaths(covidTable, 3);
+    io:println("Countries with highest deaths:", countriesWithDeaths);
+```
+
+## The completed code for sorting
+
+Below is the complete code for the sorting function.
 
 ```ballerina
+public function filterCountriesByCases(table<CovidEntry> dataTable, decimal noOfCases) returns string[] {
+    string[] filteredCountries = from CovidEntry entry in dataTable
+        where entry.cases > noOfCases
+        select entry.country;
+    return filteredCountries;
+}
+
+public function findCountriesByHighestNoOfDeaths(table<CovidEntry> dataTable, int n) returns [string, decimal][] {
+    [string, decimal][] countriesWithDeaths = from CovidEntry entry in dataTable
+        order by entry.deaths descending
+        limit n
+        select [entry.country, entry.deaths];
+    return countriesWithDeaths;
+}
+
```

Comment:
This is the only part related to sorting. Can we highlight this part only and say, below is the complete code after adding the sorting logic which is highlighted.?

Reply:
Fixed.